WordPress/src/main/java/org/wordpress/android/viewmodel/mlp/ModalLayoutPickerViewModel.kt

```diff
@@ -34,28 +46,62 @@ class ModalLayoutPickerViewModel @Inject constructor(
     private val _uiState: MutableLiveData<UiState> = MutableLiveData()
     val uiState: LiveData<UiState> = _uiState
 
+    private val contentUiState: ContentUiState
+        get() = uiState.value as? ContentUiState ?: ContentUiState()
+
     /**
      * Create new page event
      */
     private val _onCreateNewPageRequested = SingleLiveEvent<Unit>()
     val onCreateNewPageRequested: LiveData<Unit> = _onCreateNewPageRequested
 
+    init {
+        dispatcher.register(this)
+    }
+
+    override fun onCleared() {
+        dispatcher.unregister(this)
+        super.onCleared()
+    }
+
     fun init() {
-        updateUiState(ContentUiState())
-        loadLayouts()
-        loadCategories()
-        updateButtonsUiState()
+        fetchLayouts()
+    }
+
+    private fun fetchLayouts() {
```

Comment:
We might want to consider moving this to a bg/io thread, wdyt?

Reply:
I agree. Fixed with 05ca9af80c966cdba96a30b4335a4a04bf31901e

Reply:
It seems the code is still running on the main thread. We invoke `updateUiState(state.copy(layoutCategories = listItems))` which directly sets value to uiState livedata and the app doesn't crash. Since we extend `ScopedViewModel` I think we can drop `viewModelScope` before the `launch`. We'll need to wrap "updateUiState" with "withContext(main)". Wdyt?

Reply:
You are right. Please check b3fc5e3e39f01fa443c7d3264b28da7df6da74d2